contracts/PriceLinearTime.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/metatx/ERC2771ContextUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "./interfaces/IPriceDynamic.sol";
+
+struct Linear {
+    uint64 slopeEnumerator;
+    uint64 slopeDenominator;
+    uint64 startTime;
+}
+
+/**
+ * @title PublicFundraising
+ * @author malteish, cjentzsch
+ * @notice This contract represents the offer to buy an amount of tokens at a preset price. It can be used by anyone and there is no limit to the number of times it can be used.
+ *      The buyer can decide how many tokens to buy, but has to buy at least minAmount and can buy at most maxAmount.
+ *      The currency the offer is denominated in is set at creation time and can be updated later.
+ *      The contract can be paused at any time by the owner, which will prevent any new deals from being made. Then, changes to the contract can be made, like changing the currency, price or requirements.
+ *      The contract can be unpaused after "delay", which will allow new deals to be made again.
+ *      A company will create only one PublicFundraising contract for their token (or one for each currency if they want to accept multiple currencies).
+ * @dev The contract inherits from ERC2771Context in order to be usable with Gas Station Network (GSN) https://docs.opengsn.org/faq/troubleshooting.html#my-contract-is-using-openzeppelin-how-do-i-add-gsn-support
+ */
+contract PriceLinearTime is
+    ERC2771ContextUpgradeable,
+    Ownable2StepUpgradeable,
+    ReentrancyGuardUpgradeable,
```

Comment:
Nutzen wir den ReentrancyGuard irgendwo hier?

Reply:
Oder heißt das wenn man davon erbt wir er automatisch in jeder Funktion genutzt?

Reply:
Der ... wird auch nicht gebraucht. :/

Reply:
Auch unnötig, wurde entfernt. Danke für den Hinweis!